js/content/common.js

```diff
@@ -196,16 +196,45 @@ let ExtensionLayer = (function() {
 
     // NOTE: use cautiously!
     // Run script in the context of the current tab
-    self.runInPageContext = function(fun){
-        let script  = document.createElement('script');
-        script.textContent = '(' + fun + ')();';
+    self.runInPageContext = function(fun) {
+        let script  = document.createElement("script");
+        script.textContent = '(' + fun + ")();";
         document.documentElement.appendChild(script);
         script.parentNode.removeChild(script);
     };
 
     return self;
 })();
 
+class Messenger {
+    static sendMessage(msgID, info) {
+        window.postMessage({
+            type: "es_" + msgID,
+            information: info
+        }, '*');
```

Comment:
It's good practice to *always* specify the `targetOrigin` when you know it, and we know it: `window.location.origin`

Reply:
I thought about this too, but doesn't the extension context have a different origin?